sklearn/ensemble/tests/test_forest.py

```diff
@@ -833,6 +837,21 @@ def test_sparse_input():
         yield check_sparse_input, name, X, sparse_matrix(X), y
 
 
+def check_no_sparse_y_support_classifier(name):
+    X, y = datasets.make_multilabel_classification(random_state=0,
+                                                   n_samples=50)
+    y_sparse = csr_matrix(y)
+    ForestClassifier = FOREST_CLASSIFIERS[name]
+    assert_raise_message(ValueError, "Unknown label type: 'unknown'",
```

Comment:
I don't think that's a useful error message. For sparse X not supported we raise TypeError (not that I'm sure this decision was correct) with message "A sparse matrix was passed, but dense data is required. Use X.toarray() to convert to a dense numpy array."

One option is to use `check_X_y` and extend upon its `multioutput` parameter with an `allow_sparse_y` parameter to standardise checking and error message.